src/codegen/generate.ts

```diff
@@ -1215,57 +1212,121 @@ export default class ApiGenerator {
           return name;
         };
 
-        // build the method signature - first all the required parameters
-        const methodParams = required.map((p) =>
-          cg.createParameter(getArgName(this.resolve(p)), {
-            type: this.getTypeFromParameter(p),
-          }),
-        );
-
-        let body: OpenAPIV3.RequestBodyObject | undefined;
-        let bodyVar;
+        const methodParams: ts.ParameterDeclaration[] = [];
+        let body: OpenAPIV3.RequestBodyObject | undefined = undefined;
+        let bodyVar: string | undefined = undefined;
+        switch (this.opts.argumentStyle ?? "positional") {
+          case "positional":
```

Comment:
Since the `generateApi` method already is quite convoluted. Can we refactor the body of this switch to standalone methods/functions such as `getPositionalMethodParams` and `getObjectMethodParams`?